### ISSUE TITLE
Silently failing to Rpathify/auto-map in build_tarballs.jl

### ISSUE THREAD

**Issue by juan-pablo-vielma** [JuliaPackaging/BinaryBuilder.jl#332]
The following loop in `check_dynamic_linkage` will be interrupted any time a library cannot be auto-mapped (e.g. line 266). 
https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/1fee900486baedfce66ddb24872133ef36b9d899/src/Auditor.jl#L225-L288

By default the result of `check_dynamic_linkage` is ignored by
https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/1fee900486baedfce66ddb24872133ef36b9d899/src/Dependency.jl#L77-L79
and
https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/1fee900486baedfce66ddb24872133ef36b9d899/src/AutoBuild.jl#L391
does not set `ignore_audit_errors=false` so this is silently ignored when building tarballs, which can cause linking errors later. The errors can be tricky to track if `length(keys(libs))>1` in  `check_dynamic_linkage` and an early library has trouble and interrupts the loop preventing other libraries without any trouble to be auto-mapped. For example, I encountered a failure to generate a `deps.jl` caused by 
```julia
julia> Libdl.dlopen("libCoinUtils")
ERROR: could not load library "libCoinUtils"
dlopen(libCoinUtils.dylib, 1): Library not loaded: /workspace/destdir/lib/libcoinlapack.1.dylib
  Referenced from: /Users/jvielma/.julia/v0.6/Clp/deps/usr/lib/libCoinUtils.3.10.14.dylib
  Reason: image not found
```
but `build_tarballs.jl` just reported
```julia
Info: Checking lib/libCoinUtils.3.10.14.dylib with RPath list String[]
Warning: Linked library /usr/lib/libbz2.1.0.dylib could not be resolved and could not be auto-mapped
```

Looks like libbz2 would not have caused a problem (see below), so maybe removing the three `return false` from the loop (and using an `all_ok` line in `audit()`) makes more sense than setting `ignore_audit_errors=false`?

```bash
han:lib jvielma$ otool -L libCoinUtils.3.dylib 
libCoinUtils.3.dylib:
	/workspace/destdir/lib/libCoinUtils.3.dylib (compatibility version 14.0.0, current version 14.14.0)
	/usr/lib/libbz2.1.0.dylib (compatibility version 1.0.0, current version 1.0.5)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.5)
	/workspace/destdir/lib/libcoinglpk.1.dylib (compatibility version 12.0.0, current version 12.5.0)
	/workspace/destdir/lib/libcoinlapack.1.dylib (compatibility version 7.0.0, current version 7.6.0)
	/workspace/destdir/lib/libcoinblas.1.dylib (compatibility version 6.0.0, current version 6.6.0)
	/opt/x86_64-apple-darwin14/x86_64-apple-darwin14/lib/libgfortran.4.dylib (compatibility version 5.0.0, current version 5.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1213.0.0)
	/opt/x86_64-apple-darwin14/x86_64-apple-darwin14/lib/libquadmath.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.0.0)
```

**Comment by juan-pablo-vielma** [comment]
Quick update. Removing the `return false` (see below) does result on the correct Rpathification:
```julia
JVIELMA1:lib jvielma$ otool -L libCoinUtils.dylib 
libCoinUtils.dylib:
	/workspace/destdir/lib/libCoinUtils.3.dylib (compatibility version 14.0.0, current version 14.14.0)
	/usr/lib/libbz2.1.0.dylib (compatibility version 1.0.0, current version 1.0.5)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.5)
	@rpath/libcoinglpk.1.dylib (compatibility version 12.0.0, current version 12.5.0)
	@rpath/libcoinlapack.1.dylib (compatibility version 7.0.0, current version 7.6.0)
	@rpath/libcoinblas.1.dylib (compatibility version 6.0.0, current version 6.6.0)
	@rpath/libgfortran.4.dylib (compatibility version 5.0.0, current version 5.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1213.0.0)
	@rpath/libquadmath.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.0.0)
```
However, I do get two new (inconsequential?) errors
```julia
[10:08:04] error: /opt/x86_64-apple-darwin14/bin/install_name_tool: for: lib/libCoinUtils.3.10.14.dylib (for architecture x86_64) option "-add_rpath @loader_path/." would duplicate path, file already has LC_RPATH for: @loader_path/.
...
[10:08:04] error: /opt/x86_64-apple-darwin14/bin/install_name_tool: for: lib/libCoinUtils.3.10.14.dylib (for architecture x86_64) option "-add_rpath @loader_path/." would duplicate path, file already has LC_RPATH for: @loader_path/.
```
so who knows what else I am doing wrong :-) 

With the elimination of the `return false` a rather horrible build of Clp (https://github.com/juan-pablo-vielma/Clp.jl/blob/bbv0.4.0/deps/build.jl) does succeed in Linux, Mac and Windows 10. Only one of the binary dependencies needed the update. 

Just in case here is my update. Skipped the PR because it is rather trivial and I am still not sure if this is the correct solution. Happy to submit it if would help. 

https://github.com/juan-pablo-vielma/BinaryBuilder.jl/blob/7008ed372a32c56cccf9902c416214feeda8ca4b/src/Auditor.jl#L225-L296

**Comment by juan-pablo-vielma** [comment]
Fixed here 92cd344262e6e13475334733f9f1b7325290258d
